Engine/platform/windows/gfx/ali3dd3d.cpp

```diff
@@ -734,7 +734,7 @@ int D3DGraphicsDriver::_initDLLCallback(const DisplayMode &mode)
 
 void D3DGraphicsDriver::InitializeD3DState()
 {
-  direct3ddevice->Clear(0, NULL, D3DCLEAR_TARGET, D3DCOLOR_RGBA(30, 0, 0, 255), 0.5f, 0);
```

Comment:
Should we set the viewport here as well? I had a thought that maybe in debug mode we should set this to a non-black colour to make it obvious of rendering issues.

Reply:
I thought the "30" was unintentional. Want me to put it back, or change the color?

Reply:
It's probably fine the way you have it. I'm not actually sure this line has any effect! 

As an experiment, I changed the clear colour to bright red, initialised the screen viewport earlier, disabled the Clear() call in `_render` and moved the Clear to later in the initialisation. I expected to see a left over red borders but it was just default black (with a flash of red where we render the game).

I'm not sure what I'm missing here (not a direct3d expert), but I could not find a way to clear the screen to a colour during Initialisation before we do Render calls. 

Although I do see some funky stuff if I toggle between full screen and windowed multiple times. (with the changes for the experiment)